### DISCUSSION TITLE
RP2040 Fun Tips

### DISCUSSION THREAD

**Discussion by SezaroSteve** [ZuluSCSI/ZuluSCSI-firmware#134] [Ideas]
Fun Tip 1:
I use the RP2040 with a "SMC Beleth's Drum, HD sound simulator" from Infinity Products in Japan. It produces a sound that is designed to mimic the old Quantum LPS Series of SCSI Harddisks and it works very well. One really gets the impression that there is one of them old school HDD's in there. Great solution.
The Beleth's Drum is hard to come by though. I am therefore currently exploring using the new "normal / full.size" RP2040's speaker-header to produce a similar audio effect.

Fun Tip 2:
In the zuluscsi.ini file, one can override the vendor and product strings a SCSI controller reads. So normally, it would say ZULUSCSI etc.
But what I do, for a bit of nostalgia and old sentiment, is use the "vendor", "product" and "version" parameters of individual (emulated) SCSI Devices to override the default so that the host OS "sees" a real period-correct Quantum or Seagate SCSI Harddisk in there.
This is the relevant part of the zuluscsi.ini that I use in this particular Amiga 1200:

```
[SCSI0]
Vendor = "QUANTUM"
Product = "LPS270S"
Version = 0E4

[SCSI1]
Vendor = "QUANTUM"
Product = "ST21S012"
Version = 02-C

[SCSI2]
Vendor = "Seagate"
Product = "ST3600N"
Version = ZD32

[SCSI6]
Vendor = "Pioneer"
Product = "DR-506S"
Type = 2
```

The Amiga thinks there is a real Quantum LPS270S with Firmware 0E4 in there. This was a 200MB Drive back in the day and that firmware actually existed. So optically, when using a tool to display devices in the system, it looks totally real :-)
Same goes for the second HDD, a Quantum ST21S012 with Firmware 02-C. This was a 2GB, single-ended 50-pin SCSI-2 HDD. Also period correct.
The third HDD, is a "fake" Seagate ST3600N with Firmware ZD32. I present this HDD to the ShapeShifter Amiga MacOS Simulator and MacOS 7.x and 8.x think there is actually a real Seagate HDD in the system. This make and model and firmware actually existed in the Apple Quadra 900 computer which ShapeShifter emulates.
The last entry is a period correct "Pioneer DR-506S" slot-loading SCSI-2 CD-ROM drive. I could not find firmware version of it so I left that blank.

Tip 3:
When using HDD Disk-images ("HD0_512.hda" etc.), these files are normal files for a Windows, Linux or Mac PC. So creating backups of your Amiga or Mac or sound-sampler emulated Hard-disks, is as easy as removing the SD-Card from the ZuluSCSI, sticking it into the PC and copying the contents of the card to your PC.
When something goes wrong (I totally corrupted a virtual HDD presented to MacOS via Amiga ShapeShifter when I enabled "SCSI Direct Transfers" in ShapeShifter), you take the card out, put it in your PC and copy back the previous version of the HDD image file. It really is as simple as that.

**Comment by Br0ther66** [comment]
Great tips! I'm going to put Tip 2 into action, which I really liked.

Thanks!!
